sdk/src/main/java/co/omise/android/models/Source.kt

```diff
@@ -0,0 +1,176 @@
+package co.omise.android.models
+
+import android.os.Parcel
+import android.os.Parcelable
+import co.omise.android.api.Endpoint
+import co.omise.android.api.RequestBuilder
+import com.fasterxml.jackson.annotation.JsonProperty
+import okhttp3.HttpUrl
+import okhttp3.RequestBody
+import java.io.IOException
+
+/**
+ * Represents Source object and contains its [RequestBuilder].
+ *
+ * @see [Sources API](https://www.omise.co/sources-api)
+ */
+data class Source(
+        var type: SourceType? = null,
+        var flow: FlowType? = null,
+        var amount: Long = 0,
+        var currency: String? = null,
+        var barcode: String? = null,
+        var references: References? = null,
+        @field:JsonProperty("store_id")
```

Comment:
Not sure if this is a good idea to have properties of every possible source type in one Source class. How’s about subclassing?

Reply:
While I agree that when I looked at this at the beginning, it's very messy and not optimal and having it as individual subclasses would certainly help with this. 
But having said that, I can also see how doing that would make the code more complex and less flexible and that could be the reason we have kept it that way in `omise-java`. Once I'm back next week, we can discuss on what's the best approach for improving this.

Reply:
That's just my suggestion. I think we better stick to the convention of the language. Could you check how normally Kotlin devs do this similar things?

Reply:
Yeah I understand, I'm just saying I'd like us to take a little bit of time to figure out how to best improve this.

Reply:
OK, let's discuss about this later

Reply:
🙇